src/librustc/ty/mod.rs

```diff
@@ -897,6 +897,28 @@ pub enum Predicate<'tcx> {
     ConstEvaluatable(DefId, &'tcx Substs<'tcx>),
 }
 
+/// The crate outlives map is computed during typeck and contains the
+/// outlives of every item in the local crate. You should not use it
+/// directly, because to do so will make your pass dependent on the
+/// HIR of every item in the local crate. Instead, use
+/// `tcx.inferred_outlives_of()` to get the outlives for a *particular*
+/// item.
+pub struct CratePredicatesMap {
+    //todo
+    //    /// This relation tracks the dependencies between the variance of
+    //    /// various items. In particular, if `a < b`, then the variance of
+    //    /// `a` depends on the sources of `b`.
+    //    pub dependencies: TransitiveRelation<DefId>,
```

Comment:
we should not need this; in fact, we can remove it from variance too, but that's a separate issue I suppose